tests/queries/0_stateless/03271_s3_table_function_asterisk_glob.sql

```diff
@@ -0,0 +1,12 @@
+-- Tags: no-parallel, no-fasttest
+-- Tag no-fasttest: Depends on AWS
+
+SET max_threads = 1;
+SET s3_truncate_on_insert = 1;
+
+INSERT INTO FUNCTION s3(s3_conn, filename='dir1/03271_s3_table_function_asterisk_glob/', format=Parquet) SELECT 0 as num;
```

Comment:
https://github.com/ClickHouse/ClickHouse/pull/72007 would break this test by no longer including paths with empty file names when using a trailing wildcard (`/*`). Is that ok or is this behavior needed for a specific use case?

Reply:
i don't have any use case in mind, it just seems to be the correct behaviour. because s3 is not a fs, object key is a single string that might end on a '/'. initially i was asked by Pete why query with `/dir/*` pattern didn't work and it turned out that he had an object with name `dir/`, we got it in the `List` output, found that its name is empty and also throw away all other objects. so i decided to try to fix both problems - don't give up when we see an "empty file name" and read from such objects too.

Reply:
I was solving the same original problem with  #72007. :-)

> don't give up when we see an "empty file name" and read from such objects too

Makes sense to read from it when there is data in it, but shouldn't we rather skip it if it's empty? 

Because Amazon creates such objects when you create a directory in the S3 console and it doesn't really make sense to try to read from such objects (from [here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-folders.html)):

> When you create a folder in Amazon S3, S3 creates a 0-byte object with a key that's set to the folder name that you provided. For example, if you create a folder named photos in your bucket, the Amazon S3 console creates a 0-byte object with the key photos/. The console creates this object to support the idea of folders.

Reply:
thx for this reference - i was wondering who creates this empty objects, seemed hard to believe that people really do it manually.